salt/pillar/netbox.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+'''
+A module that adds data to the Pillar structure from a NetBox API.
+
+
+Configuring the NetBox ext_pillar
+====================================
+
+.. code-block:: yaml
+
+  ext_pillar:
+    - netbox:
+        api_url: http://netbox_url.com/api/
+        api_token: 123abc
+
+Create a token in your NetBox instance at
+http://netbox_url.com/user/api-tokens/
+
+The following options are optional, and determine whether or not
+the module will attempt to configure the ``proxy`` pillar data for
+use with the napalm proxy-minion:
+
+.. code-block:: yaml
+
+        proxy_return: True
+        proxy_username: admin
+
+By default, this module will query the NetBox API for the platform
+associated with the device, and use the 'NAPALM driver' field to
+set the napalm proxy-minion driver. (Currently only 'napalm' is supported
+for drivertype.)
+
+This module currently only supports the napalm proxy minion and assumes
+you will use SSH keys to authenticate to the network device.  If password
+authentication is desired, it is recommended to create another ``proxy``
+key in pillar_roots (or git_pillar) with just the ``passwd`` key and use
+:py:func:`salt.renderers.gpg <salt.renderers.gpg>` to encrypt the value.
+If any additional options for the proxy setup are needed they should also be
+configured in pillar_roots.
+'''
+
+
+from __future__ import absolute_import, print_function, unicode_literals
+import logging
+
+try:
+    import requests
+    import ipaddress
+    _HAS_DEPENDENCIES = True
+except ImportError:
+    _HAS_DEPENDENCIES = False
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    return _HAS_DEPENDENCIES
+
+
+def ext_pillar(minion_id, pillar, *args, **kwargs):
+    '''
+    Query NetBox API for minion data
+    '''
+
+    # Pull settings from kwargs
+    api_url = kwargs['api_url'].rstrip('/')
+    api_token = kwargs['api_token']
```

Comment:
I believe the token should not be mandatory, as this Pillar module only fetches data (the token is required only when writing data into NetBox) - see https://docs.saltstack.com/en/develop/ref/modules/all/salt.modules.netbox.html as an example.